component/main.jsonnet

```
@@ -9,18 +9,48 @@ local instance = inv.parameters._instance;
 
 local common = import 'common.libsonnet';
 
+local instanceStacks = std.mapWithKey(
+  function(name, param) common.stackForInstance(name),
+  params.instances
+);
+local mergedParams = std.mapWithKey(
+  function(name, param) params.base + com.makeMergeable(param),
+  params.instances
+);
+
+
+local namespacesPromLabels =
+  local f(prev, i) =
+    local p = mergedParams[i];
+    local stack = instanceStacks[i];
+    prev {
+      [if p.prometheus.enabled then stack.values.prometheus.namespace]+: {
+        ['monitoring.syn.tools/%s' % i]: 'true',
+      },
+    };
+  std.foldl(f, std.objectFields(params.instances), {});
+
 local namespaces = std.foldl(
   function(namespaces, nsName)
     if params.namespaces[nsName] != null then
-      namespaces { ['00_namespace_%s' % nsName]: kube.Namespace(nsName) + {
-        metadata+: com.makeMergeable(params.namespaces[nsName]),
-      } }
+      namespaces {
+        ['00_namespace_%s' % nsName]:
+          kube.Namespace(nsName)
+          +
+          {
+            metadata+: {
+              labels+: com.getValueOrDefault(namespacesPromLabels, nsName, {}),
+            },
+          }
+          + {
+            metadata+: com.makeMergeable(params.namespaces[nsName]),
+          },
```

Is this intentionally implemented like this, so users can override the auto-generated value for the `monitoring.syn.tools/<instance>` label in the namespaces parameter?

Otherwise you could probaby do

```suggestion
          {
            metadata+: com.makeMergeable(params.namespaces[nsName]) {
              labels+: com.getValueOrDefault(namespacesPromLabels, nsName, {}),
            },
          },
```

---

I did do this intentionally with the reasoning, if they really wanted to they can. Not entirely sure if that's a good idea or not

---

IMO it's fine to leave it like that for now.